usecase/src/test/java/org/mobilitydata/gtfsvalidator/usecase/ValidateNoOverlappingStopTimeInTripBlockTest.java

```diff
@@ -0,0 +1,1523 @@
+        // |---------|--------|-----------|---------|
+        // | 0       | 2      | a         | 7       |
+        // | 0       | 5      | b         | 7       |
+        // | 0       | 8      | c         | 7       |
+
+        // stop_times.txt
+        // | tripId | arrivalTime | departureTime | stopId | stopSequence  |
+        // |--------|-------------|---------------|--------|---------------|
+        // | 2      | 07:00       | 07:40         | 101    | 10            |
+        // | 2      | 08:20       | 10:00         | 102    | 11            |
+        // | 5      | 11:03       | 11:15         | 103    | 12            |
+        // | 5      | 11:16       | 11:20         | 104    | 13            |
+        // | 8      | 09:40       | 10:20         | 105    | 14            |
+        // | 8      | 10:30       | 13:50         | 106    | 15            |
+
+        // trips.txt + stop_times.txt
+        // | routeId | tripId | serviceId | blockId | first stop time | last stop time |
+        // |---------|--------|-----------|---------|-----------------|----------------|
+        // | 0       | 2      | a         | 7       | 07:00           | 10:00          |
+        // | 0       | 5      | b         | 7       | 11:03           | 11:20          |
+        // | 0       | 8      | c         | 7       | 09:40           | 13:50          |
+
+        // calendar.txt: file not provided
+
+        // calendar_dates.txt:
+        // | serviceId | date     | exception_type |
+        // |-----------|----------|----------------|
+        // | a         | 20200723 | 1              |
+        // | a         | 20200801 | 1              |
+        // | b         | 20200904 | 0              |
+        // | c         | 20200723 | 1              |
+        // Here, services a and c overlap: they are both active on 2020-07-23.
+        // Service a starts at 07:00 and ends at 10:00 - and service c starts at 09:40 and ends at 13:50.
+        // Which means that trips 2 and 8 overlap on 2020-07-23 between 09:40 and 10:00: that is an error and should
+        // generate a notice.
+
+        final GtfsDataRepository mockDataRepo = mock(GtfsDataRepository.class);
+        final Map<String, List<Trip>> mockTripPerBlockIdCollection = new HashMap<>();
+        final List<Trip> mockTripCollection = new ArrayList<>();
+
+        final Trip firstMockTrip = mock(Trip.class);
+        when(firstMockTrip.getTripId()).thenReturn("2");
+        when(firstMockTrip.getServiceId()).thenReturn("a");
+        when(firstMockTrip.getBlockId()).thenReturn("7");
+        when(firstMockTrip.hasSameServiceId(any())).thenReturn(false);
+
+        final Trip secondMockTrip = mock(Trip.class);
+        when(secondMockTrip.getTripId()).thenReturn("5");
+        when(secondMockTrip.getServiceId()).thenReturn("b");
+        when(secondMockTrip.getBlockId()).thenReturn("7");
+        when(secondMockTrip.hasSameServiceId(any())).thenReturn(false);
```

Comment:
Same on not mocking domain class methods

Reply:
As discussed, implementation of domain layer classes is tested separately, so that tests of use case layer only verify interactions with mocked objects.